src/decode.rs

```diff
@@ -135,16 +151,16 @@ impl<S: Stream> Decoded<S> {
     //
     // Returns the event for dispatch if it is complete.
     fn parse_complete_lines_into_event(mut self: Pin<&mut Self>) -> Result<Option<Event>> {
-        trace!("***** we are going to start the parsing process");
         let mut seen_empty_line = false;
 
         let this = self.as_mut().project();
 
         while let Some(line) = this.complete_lines.pop_front() {
-            trace!("***** initial line received is {:?})", line);
-            if line.is_empty() {
+            if line.is_empty() && this.event.is_some() {
```

Comment:
Fixes an issue with multiple line terminators between events within a single chunk. There is a unit test for this added below.